pkg/visibility/api/pending_workload_CQ.go

```
@@ -0,0 +1,112 @@
+var _ rest.Scoper = &pendingWorkloadsInCQ{}
+var _ rest.TableConvertor = &pendingWorkloadsInCQ{}
+var _ rest.SingularNameProvider = &pendingWorkloadsInCQ{}
+
+func newPendingWorkloadsInCQ(kueueMgr *queue.Manager) *pendingWorkloadsInCQ {
+	return &pendingWorkloadsInCQ{
+		kueueMgr: kueueMgr,
+		log:      ctrl.Log.WithName("pending-workload-cq"),
+	}
+}
+
+// New implements rest.Storage interface
+func (m *pendingWorkloadsInCQ) New() runtime.Object {
+	return &v1alpha1.PendingWorkloadSummary{}
+}
+
+// Destroy implements rest.Storage interface
+func (m *pendingWorkloadsInCQ) Destroy() {
+}
+
+// Kind implements rest.KindProvider interface
+func (m *pendingWorkloadsInCQ) Kind() string {
+	return "PendingWorkloadsInClusterQueue"
+}
+
+// NewList implements rest.Lister interface
+func (m *pendingWorkloadsInCQ) NewList() runtime.Object {
+	return &v1alpha1.PendingWorkloadSummaryList{}
+}
+
+// List implements rest.Lister interface
+func (m *pendingWorkloadsInCQ) List(ctx context.Context, options *metainternalversion.ListOptions) (runtime.Object, error) {
+	// TODO: consider returning error
+	return &v1alpha1.PendingWorkloadSummaryList{}, nil
+}
+
+// Get implements rest.Getter interface
+func (m *pendingWorkloadsInCQ) Get(ctx context.Context, name string, opts *metav1.GetOptions) (runtime.Object, error) {
+	m.kueueMgr.UpdateSnapshot(name, WholeSnapshot)
+	var wls []v1alpha1.PendingWorkloadSummary
+	for _, val := range m.kueueMgr.GetSnapshot(name) {
+		wls = append(wls, v1alpha1.PendingWorkloadSummary{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      val.Name,
+				Namespace: val.Namespace,
+			},
+		})
+	}
+	return &v1alpha1.PendingWorkloadSummaryList{Items: wls}, nil
+}
+
+// ConvertToTable implements rest.TableConvertor interface
+func (m *pendingWorkloadsInCQ) ConvertToTable(ctx context.Context, object runtime.Object, tableOptions runtime.Object) (*metav1beta1.Table, error) {
+	var table metav1beta1.Table
+	return &table, nil
```

I would also return an error here, if it is not needed.